### PR TITLE
IP: add is_device_up

### PR DESCRIPTION
**Pull request by mcgov**
Add a function to check current device status using `ip -br link show`

### CODE REVIEW

File: lisa/tools/ip.py [hunk 1]
```diff
@@ -83,6 +83,9 @@ class Ip(Tool):
         r"\s+dev\s+"  # looking for the device for the default route
         r"([a-zA-Z0-9]+)"  # capture device
     )
+    __ip_br_show_regex = re.compile(
```
Comment (squirrelsc) [inline lisa/tools/ip.py:89]: Move examples above the regexp.

File: lisa/tools/ip.py [hunk 2]
```diff
@@ -127,6 +133,20 @@ def _get_matched_dict(self, result: str) -> Dict[str, str]:
             "ip_addr": matched.group("ip_addr"),
         }
 
+    def is_device_up(self, nic_name: str) -> bool:
+        device_info = self.node.execute(
+            f"ip -br link show {nic_name}",
```
Comment (LiliDeng) [inline lisa/tools/ip.py:138]: put it into ip tool

Comment (mcgov) [reply lisa/tools/ip.py:138]: Not sure what you mean, pull the regex into the tool?
or use self.run?

File: lisa/tools/ip.py (hunk 1 above)
Comment (mcgov) [reply lisa/tools/ip.py:89]: moved

**Comment by mcgov** [comment]
closing, testing this without the other rdma-core PR is getting too complicated

**Pull request closed without merging**